client/app/expenses/components/PaymentSummary.tsx

```diff
@@ -0,0 +1,102 @@
+import { BalanceSummary } from '../types'
+
+type PaymentSummaryProps = {
+  balanceSummary: BalanceSummary[]
+  userId: number
+  year: number
+  month: number
+}
+
+const PaymentSummary = ({
+  balanceSummary,
+  userId,
+  year,
+  month,
+}: PaymentSummaryProps) => {
+  const { onUpdate: updatePaymentsStatus, isPending } =
+    useUpdatePaymentsStatus()
+
+  const houseId = useGetHouseInfo().houseId
+
+  const handleUpdatePaymentSummaryStatus = (
+    payerId: number,
+    payeeId: number,
+    paidDate: null | string,
+  ) => {
+    const data = {
+      year,
+      month,
+      buyerId: payeeId,
+      payerId,
+      isPaid: paidDate === null ? true : false,
+      houseId,
+    }
+    updatePaymentsStatus(data)
+  }
+
+  return (
+    <div className='w-full rounded-sm border-[1px] border-slate-400 bg-slate-100 px-4 py-4 md:w-2/3 md:px-8 md:py-4'>
+      <div className='flex items-center justify-start border-b-4 border-double border-slate-300 pb-2 '>
+        <p className='text-base font-bold  md:text-lg'>Payment Summary</p>
+      </div>
+      <div className='flex flex-col gap-4 pt-4'>
+        {balanceSummary.map((payer) => (
+          <div
+            className='flex items-start justify-center gap-4 border-b-[1px] border-dotted border-slate-400 pb-2 last:border-0'
+            key={payer.payerId}
+          >
+            <div className='flex min-w-[80px] items-center gap-2 md:min-w-[130px]'>
+              <UserDisplay
+                name={payer.payerName}
+                icon={payer.payerIcon}
+                size={28}
+              />
+            </div>
+            <div className='flex items-center pt-1'>
+              <FiArrowRight size={16} />
+            </div>
+            <div className='flex w-full flex-col gap-2'>
+              {payer.payees.map((payee) => (
+                <div
+                  className='flex items-center justify-between gap-4'
+                  key={payee.payeeId}
+                >
+                  <div className='flex items-start justify-center gap-4 '>
+                    <Checkbox
+                      className='h-6 w-6'
+                      checked={payee.paidDate === null ? false : true}
+                      disabled={
+                        isPending ||
+                        (payer.payerId !== userId && payee.payeeId !== userId)
+                      }
+                      onClick={() =>
+                        handleUpdatePaymentSummaryStatus(
+                          payer.payerId,
+                          payee.payeeId,
+                          payee.paidDate,
+                        )
+                      }
+                    />
+                    <UserDisplay
+                      name={payee.payeeName}
+                      icon={payee.payeeIcon}
+                      size={28}
+                    />
+                  </div>
+                  <span className='font-medium md:text-lg'>$ {payee.fee}</span>
```

Comment:
If the user name is long, it seems to be broken! 
<img width="386" alt="Screenshot 2024-04-11 at 2 16 36 PM" src="https://github.com/HAK2024/fairy-share/assets/66394413/79b5f6a6-9d40-4bde-ba37-07360aea354a">

Reply:
Sorry same as perDate!

<img width="388" alt="Screenshot 2024-04-11 at 2 12 46 PM" src="https://github.com/HAK2024/fairy-share/assets/66394413/106c22dd-f077-4b2b-90d7-56e210e16562">

Reply:
Fixed!
https://github.com/HAK2024/fairy-share/pull/119/commits/374de6e70d68c80eea90648228d3d2bd761d1dac

Reply:
Still looks broken!

PC: monthly payment
<img width="724" alt="Screenshot 2024-04-11 at 5 04 52 PM" src="https://github.com/HAK2024/fairy-share/assets/66394413/29edcda3-703b-4636-b3cf-afcbb32ba0a1">

SP: monthly payment
<img width="412" alt="Screenshot 2024-04-11 at 5 04 40 PM" src="https://github.com/HAK2024/fairy-share/assets/66394413/ca8ebfe4-3e31-47b9-9d30-6115bfb548ec">

Also, it's better to make the fontsize of user name to 14px instead of 16px!
<img width="396" alt="Screenshot 2024-04-11 at 5 03 53 PM" src="https://github.com/HAK2024/fairy-share/assets/66394413/7aa48b62-ebf1-4c30-a2ce-cd3e92df11aa">